### PR TITLE
[RHPAM-1387 ] - Allow set Smart router location to external URL in RHPAM

### PR DESCRIPTION
**Pull request by spolti**
Follow up changes, make the smartrouter protocol and external route url
configurable.

Signed-off-by: Filippe Spolti <fspolti@redhat.com>

### CODE REVIEW

**Comment by spolti** [comment]
All, after this PR is approved, I'll cherry pick the changes for master branch.

File: pkg/controller/kieapp/defaults/defaults_test.go [hunk 1]
```diff
@@ -305,6 +304,32 @@ func TestRhpamProdwSmartRouter(t *testing.T) {
...
+
+	assert.Nil(t, err, "Error getting prod environment")
+	assert.False(t, env.SmartRouter.Omit, "SmarterRouter should not be omitted")
+	assert.Equal(t, "test-smartrouter", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_ROUTER_SERVICE"), "Variable should exist")
+	assert.Equal(t, "9443", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_ROUTER_PORT"), "Variable should exist")
+	assert.Equal(t, "https", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_ROUTER_PROTOCOL"), "Variable should exist")
+	assert.Equal(t, "test-rhpamcentrmon", env.Console.DeploymentConfigs[0].ObjectMeta.Name)
+	assert.Equal(t, "test-smartrouter", env.SmartRouter.DeploymentConfigs[0].ObjectMeta.Name)
+	assert.Equal(t, fmt.Sprintf("rhpam%s-businesscentral-monitoring-openshift", cr.Spec.CommonConfig.Version), env.Console.DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Image)
+}
```
Comment (tchughesiv) [inline pkg/controller/kieapp/defaults/defaults_test.go]: maybe check that KIE_SERVER_ROUTER_ROUTE_NAME is not set by default?

File: deploy/crs/kieapp_rhpam_production.yaml [hunk 2]
```diff
@@ -7,3 +7,5 @@ spec:
   objects:
     smartRouter:
       replicas: 1
+      useExternalRoute: false
```
[the remark this answers is not in the document]
Comment (spolti) [reply deploy/crs/kieapp_rhpam_production.yaml]: right, if not set the default is false, I forget to remove it.

**Comment by tchughesiv** [comment]
@spolti looks like the format test failed. ensure you have gofmt installed on your system and run `make test` again.

File: pkg/apis/app/v1/kieapp_types.go [hunk 3]
```diff
@@ -127,6 +127,12 @@ type KieServerSet struct {
 	Jms                 *KieAppJmsObject `json:"jms,omitempty"`
 }
 
+type SmartRouterSet struct {
```
Comment (ruromero) [inline pkg/apis/app/v1/kieapp_types.go]: Why not call it `SmartRouterObject`? Calling it Set might lead to think that it is a set of SmartRouters. For the case of `KieServerSet` it is indeed a set of KieServers that will be defined.

File: pkg/apis/app/v1/kieapp_types.go [hunk 4]
```diff
@@ -127,6 +127,12 @@ type KieServerSet struct {
...
+	KieAppObject     `json:"smartRouter,omitempty"`
```
Comment (ruromero) [inline pkg/apis/app/v1/kieapp_types.go]: define it inline to better access its attributes
```
KieAppObject    json:",inline"
```

File: pkg/controller/kieapp/defaults/defaults.go [hunk 5]
```diff
@@ -242,12 +242,21 @@ func getSmartRouterTemplate(cr *v1.KieApp) v1.SmartRouterTemplate {
 			template.KeystoreSecret = cr.Spec.Objects.SmartRouter.KeystoreSecret
 		}
 
+		if cr.Spec.Objects.SmartRouter.Protocol == "" {
+			template.Protocol = constants.SmartRouterProtocol
+		} else {
+			template.Protocol = cr.Spec.Objects.SmartRouter.Protocol
+		}
+
+		// if enpty defaults to false
```
Comment (ruromero) [inline pkg/controller/kieapp/defaults/defaults.go]: There's a typo with _empty_ but I think this comment is not needed as the default value for a boolean is `false`.

File: pkg/controller/kieapp/defaults/defaults_test.go [hunk 6]
```diff
@@ -305,6 +304,32 @@ func TestRhpamProdwSmartRouter(t *testing.T) {
 	assert.Equal(t, fmt.Sprintf("rhpam%s-businesscentral-monitoring-openshift", cr.Spec.CommonConfig.Version), env.Console.DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Image)
 }
 
+func TestRhpamProdwSmartRouterWithSSL(t *testing.T) {
```
Comment (ruromero) [inline pkg/controller/kieapp/defaults/defaults_test.go]: is there a typo there?
TestRhpamProd**w**SmartRouterWithSSL

File: pkg/apis/app/v1/kieapp_types.go (hunk 3 above)
Comment (spolti) [reply pkg/apis/app/v1/kieapp_types.go]: Good point :)

File: pkg/controller/kieapp/defaults/defaults_test.go (hunk 6 above)
Comment (spolti) [reply pkg/controller/kieapp/defaults/defaults_test.go]: it seems to, a extra w character.

**Comment by spolti** [comment]
Should I squash the commits before merge?

**Pull request merged**